api/v1alpha1/driver_types.go

```diff
@@ -35,31 +35,34 @@ const (
 
 type LogRotationSpec struct {
 	// MaxFiles is the number of logrtoate files
-	// +optional
+	//+kubebuilder:validation:Optional
 	MaxFiles int `json:"maxFiles,omitempty"`
 
 	// MaxLogSize is the maximum size of the log file per csi pods
-	// +optional
+	//+kubebuilder:validation:Optional
 	MaxLogSize resource.Quantity `json:"maxLogSize,omitempty"`
 
 	// Periodicity is the periodicity of the log rotation.
-	// +kubebuilder:validation:Enum=hourly;daily;weekly;monthly
-	// +optional
+	//+kubebuilder:validation:Optional
+	//+kubebuilder:validation:Enum:=hourly;daily;weekly;monthly
 	Periodicity PeriodicityType `json:"periodicity,omitempty"`
 
 	// LogHostPath is the prefix directory path for the csi log files
-	// +optional
+	//+kubebuilder:validation:Optional
```

Comment:
can you please why this is changed?

Reply:
I aligned all validation rules with the kubebuilder variant, they are semantically the same, but the kubebuilder variant also supports the opposite declaration (`required`) which the non-kubebuilder does not.  

I decided we need to be explicit everywhere and not assume a package-level default for required/optional.